### ISSUE TITLE
Parameters no longer support string representation for numeric values in LS 2.2+

### ISSUE THREAD

**Issue by ppf2** [elastic/logstash#4692]
Starting in LS 2.2.x, if you have a workers parameter set to a string representation, it now treats it as an invalid configuration and LS will not start up:

```
output
{
elasticsearch{
workers => "1"
}
}
```

The following is reported by --configtest:

```
./logstash -f generator.conf --configtest
The error reported is: 
  comparison of String with 1 failed
```

The above configuration works in versions prior to LS 2.2.  This is likely introduced as part of https://github.com/elastic/logstash/pull/4391 .  We used to accept string representation for numeric values.

This is marked as discuss currently to determine the right course of action:

1.  If we intend to support string representation for numeric values when defining parameters, then this is really a product bug when parsing/evaluating numeric parameter values.

2.  If we do not intend to support string representation for numeric values anymore starting in LS 2.2, please document this as a breaking change in documentation.

**Comment by jordansissel** [comment]
I don't remember what my intentions were when building the config parser and plugin validation stuff. I feel like "123" should mean the same as `123` (no quotes) when the setting has `:validate => :number`.  So I am in favor of @ppf2's suggestion 1 - "intend to support string representation for numeric values"

**Comment by jsvd** [comment]
I don't believe this is actually going to be a blocker ..or fixed by any of those tagged releases, bump to next version and remove blocker tag?
